api/src/controllers/post/mine.js

```diff
@@ -0,0 +1,68 @@
+const mongoose = require('mongoose');
+
+const postUtils = require('../../utils/post-utils');
+
+/**
+ * Create a new Mine record.
+ *
+ * @param {*} args
+ * @param {*} res
+ * @param {*} next
+ * @param {*} incomingObj Mine record to create
+ * @returns new Mine record
+ */
+exports.createRecord = async function (args, res, next, incomingObj) {
+  // TODO: Populate if needed.
+  const flavourFunctions = {};
+  return await postUtils.createRecordWithFlavours(args, res, next, incomingObj, this.createMaster, flavourFunctions);
+};
+
+/**
+ * Performs all operations necessary to create a master Mine record.
+ *
+ * @param {*} args
+ * @param {*} res
+ * @param {*} next
+ * @param {*} incomingObj Mine record to create
+ * @param {*} flavourIds array of flavour record _ids
+ * @returns created master Mine record
+ */
+exports.createMaster = function (args, res, next, incomingObj, flavourIds) {
+  let Mine = mongoose.model('MineBCMI');
+  let mine = new Mine();
+
+  mine._schemaName = 'MineBCMI';
+
+  // set permissions
+  mine.read = ['sysadmin'];
+  mine.write = ['sysadmin'];
+
+  // set data
+  incomingObj.name && (mine.name = incomingObj.name);
+  incomingObj.permitNumbers.length && (mine.permitNumbers = incomingObj.permitNumbers);
+  incomingObj.status && (mine.status = incomingObj.status);
+  incomingObj.commodities.length && (mine.commodities = incomingObj.commodities);
+  incomingObj.region && (mine.region = incomingObj.region);
+  incomingObj.location && (mine.location = incomingObj.location);
+  incomingObj.permittee && (mine.permittee = incomingObj.permittee);
+  incomingObj.type && (mine.type = incomingObj.type);
+  incomingObj.summary && (mine.summary = incomingObj.summary);
+  incomingObj.description && (mine.description = incomingObj.description);
+  incomingObj.links.length && (mine.links = incomingObj.links);
+
+  // Not checking value as it could be 0 which would fail the falsey check.
+  mine.tailingImpoundments = incomingObj.tailingImpoundments;
+
+  // Set meta. If the args exist then use the auth otherwise check the incoming object. This occurs if 
+  // the system is creating the record.
+  mine.addedBy = args && args.swagger.params.auth_payload.displayName || incomingObj.addedBy;
+  mine.updatedBy = args && args.swagger.params.auth_payload.displayName || incomingObj.updatedBy;
+  mine.dateAdded = new Date();
+  mine.dateUpdated = new Date();
+
+  // set data source reference
+  incomingObj.sourceSystemRef && (mine.sourceSystemRef = incomingObj.sourceSystemRef);
+  incomingObj._sourceRefId && (mine._sourceRefId = incomingObj._sourceRefId);
```

Comment:
Our other post controllers have some extra handling around the objectIds that we should continue here as well.  I can't remember exactly, but I think its to avoid any chance that mongo doesn't recognize them as ObjectIds, which messes with search queries, etc.
```
  incomingObj._sourceRefId &&
    ObjectId.isValid(incomingObj._sourceRefId) &&
    (inspection._sourceRefId = new ObjectId(incomingObj._sourceRefId));
```

Reply:
I did see that. The reason it isn't used here is the source system is Postgres, not Mongo. The ID field is a UUID, which is not a format that works with Mongo's Object ID. I think it will have to be saved as a string unless someone knows a better way.

Reply:
Ah gotcha, that makes sense.

Reply:
_sourceRefId is just a convenience property meant to keep track of the unique identifier in the source system.  It shouldn't have to be an ObjectId as mentioned by @KitArmstrong.  There is no requirement in MongoDB that all '_sourceRefId' instances be the same datatype for every single database document (This is counter to how SQL-based databases work).  The code, in it's safest/robust form would inspect the sourceSystemRef, and determine whether to wrap in an ObjectId, or treat as a string.

This is one of the issues with using Mongoose and having our schema slightly different in some instances of a schema.  It may be better to convert all _sourceRefId to string.